exporter/awsemfexporter/emf_exporter.go

```diff
@@ -56,66 +51,40 @@ type emfExporter struct {
 	collectorID   string
 }
 
-// newEmfPusher func creates an EMF Exporter instance with data push callback func
-func newEmfPusher(
-	config component.Config,
-	params exporter.CreateSettings,
-) (*emfExporter, error) {
+// newEmfExporter creates a new exporter using exporterhelper
+func newEmfExporter(config *Config, set exporter.CreateSettings) (*emfExporter, error) {
 	if config == nil {
 		return nil, errors.New("emf exporter config is nil")
 	}
 
-	logger := params.Logger
-	expConfig := config.(*Config)
-	expConfig.logger = logger
+	config.logger = set.Logger
 
 	// create AWS session
-	awsConfig, session, err := awsutil.GetAWSConfigSession(logger, &awsutil.Conn{}, &expConfig.AWSSessionSettings)
+	awsConfig, session, err := awsutil.GetAWSConfigSession(set.Logger, &awsutil.Conn{}, &config.AWSSessionSettings)
```

Comment:
why is this changing?  It seems like you are refactoring things that are unrelated to the specific change necessary for EMFv0

There are a few examples of that in this PR

Reply:
yeah i agree. if you must include unrelated changes in a single PR, then i'd appreciate comments saying that they are unrelated

Reply:
IMO - better to refactor in a different PR

Reply:
I'm fine with creating in different PR. However, its only simple line change and there are hardly and complication in it. If you are saying refactoring more than 5 files with more than 50 lines then yes, I would agree with you.